### DISCUSSION TITLE
Unity sample 2.0 - await connectData.Approval gets stuck.

### DISCUSSION THREAD

**Discussion by iulianbroasca** [WalletConnect/walletconnect-monorepo#2952] [Developer Support]
Hello,

I took your example from the 2.0 branch (unity repository), added the setup for my project, and after receiving "Connection successful, URI:" it gets stuck at _await connectData.Approval_.

The first time I tried with your project setup with your ID and fields, I noticed that it doesn't work, so I changed the setup for my project and it doesn't also.

I also tried both options scanning the QR code and deep link to Metamask. After pressing connect button and returning back to our app nothing happens.

Is there something I missed?

**Comment by ByteToHex** [comment]
Hi, did you ever manage to resolve this? I am encountering a similar issue (WalletConnector.cs will not assign the Uri).
